app/code/core/Mage/Catalog/Helper/Image.php

```diff
@@ -564,13 +564,18 @@ protected function getImageFile()
      */
     protected function parseSize($string)
     {
+        if ($string === null) {
+            return false;
+        }
+
         $size = explode('x', strtolower($string));
-        if (count($size) === 2) {
+        if (is_array($size) && count($size) === 2) {
```

Comment:
`explode` will always return an array, so no need for `is_array` check.

Reply:
done, thank you!